### PR TITLE
Jet ID + T3_KR_UOS available

### PR DESCRIPTION
**Pull request by minerva1993**
There are several changes in this request.
1. Added b tagging SF from twiki and b tagging shape SF from 
/afs/cern.ch/user/l/lwming/public/2018SFs

2. Updated jet id and b tagging WPs

3. Changed PAT2CAT test file which located in public space

4. Changed crab destination to UOS temporary. (due to KISTI server problem)

5. submitCrab3.py -> changed real data splitting, from  LumiBased to FileBased. Lumi based setting can cause srm --sendreceive-timeout issue, which is set to 1800 sec for default, since output file is too big or network is too slow

6. DeepCSV is now available.

7. Comment out MET filter for pilot production. Will be turned on in next update.

### CODE REVIEW

File: CatAnalyzer/src/BTagWeightEvaluator.cc [hunk 1]
```diff
@@ -55,15 +55,11 @@ void BTagWeightEvaluator::initCSVWeight(const bool useCSVHelper, const string bt
...
-    }
-    else if ( btagName == "mva" ) {
-      btagAlgo_ = BTAG_cMVAv2;
-      csvFileName = "cMVAv2_Moriond17_B_H.csv";
+      csvFileName = "CSVv2_94XSF_V1_B_F.csv";
     }
     else if ( btagName == "deepcsv" ) {
       btagAlgo_ = BTAG_DeepCSV;
-      csvFileName = "DeepCSV_Moriond17_B_H.csv";
+      csvFileName = "	DeepCSV_94XSF_V1_B_F.csv";
```
Comment (jhgoh) [inline CatAnalyzer/src/BTagWeightEvaluator.cc]: whitespace to be removed.

File: CatProducer/prod/crabConfig.py [hunk 2]
```diff
@@ -19,7 +19,9 @@
 # Where the output files will be transmitted to
 #config.Site.storageSite = 'T2_KR_KNU'
 #crab checkwrite --site=T3_KR_KISTI --lfn=/store/group/CAT/
-config.Site.storageSite = 'T3_KR_KISTI'
-#config.Site.storageSite = 'T3_KR_UOS'
-config.Data.outLFNDirBase = '/store/group/CAT/' 
+#config.Site.storageSite = 'T3_KR_KISTI'
+config.Site.storageSite = 'T3_KR_UOS'
+#config.Data.outLFNDirBase = '/store/group/CAT/' 
+config.Data.outLFNDirBase = '/store/user/jipark/'
```
Comment (jhgoh) [inline CatProducer/prod/crabConfig.py]: I think we (still) have to send output to KISTI by default

File: CatProducer/prod/submitCrab3.py [hunk 3]
```diff
@@ -55,7 +55,8 @@ def submitjob(requestName, dataset, globalTag, lumiMask, submit):
     if requestName:
         dataRequestName = '%s_%s'%(requestName,label)
         outputDatasetTag = '%s_%s'%(requestName,dataset.split("/")[2])
-        outLFNDirBase = '/store/group/CAT/%s'%(requestName) 
+        #outLFNDirBase = '/store/group/CAT/%s'%(requestName) 
+	outLFNDirBase = '/store/user/jipark/%s'%(requestName)
```
Comment (jhgoh) [inline CatProducer/prod/submitCrab3.py]: '/store/user/%s/%s" % (os.environ["USER"], requestName)
or do not change

File: CatProducer/python/producers/jetProducer_cfi.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 catJets = cms.EDProducer('CATJetProducer',
     src = cms.InputTag('slimmedJets'),
     rho = cms.InputTag('fixedGridRhoFastjetAll'),
-    btagNames = cms.vstring('pfCombinedInclusiveSecondaryVertexV2BJetTags','pfCombinedMVAV2BJetTags',"inclusiveCandidateSecondaryVerticesCvsL","pfCombinedCvsLJetTags","pfCombinedCvsBJetTags"),
+    btagNames = cms.vstring('pfCombinedInclusiveSecondaryVertexV2BJetTags',"pfDeepCSVDiscriminatorsJetTags:BvsAll","pfDeepCSVJetTags:probb","pfDeepCSVJetTags:probbb","pfDeepCSVDiscriminatorsJetTags:CvsB","pfDeepCSVDiscriminatorsJetTags:CvsL","pfCombinedCvsLJetTags","pfCombinedCvsBJetTags"),
```
Comment (jhgoh) [inline CatProducer/python/producers/jetProducer_cfi.py:6]: I think "pfDeepCSVDiscriminatorsJetTags:BvsAll" will not work at all. the right place to add them is "flavTagLabels".

File: CatAnalyzer/python/csvWeights_cfi.py [hunk 5]
```diff
@@ -5,6 +5,15 @@
...
+    csvSFLF = cms.string("csv_rwt_fit_lf_v0_final_2018_2_13.root"),
+)
+
+deepcsvWeights = cms.EDProducer("CSVWeightProducer",
+    src = cms.InputTag("catJets"),
+    leptons = cms.InputTag(""), ## Selected leptons to apply overlap removal
+    minPt = cms.double(20),
+    maxEta = cms.double(2.41),
+    deepcsvSFHF = cms.string("Deepcsv_rwt_fit_hf_v2_final_2018_2_12test.root"),
+    deepcsvSFLF = cms.string("Deepcsv_rwt_fit_lf_v2_final_2018_2_12test.root"),
```
Comment (jhgoh) [inline CatAnalyzer/python/csvWeights_cfi.py]: You will see warning messages that you have to add only one module in the cfi files.
Give different name such as csvWeights_cff.py or add new file deepcsvWeights_cfi.py and make them to contain only one module for each.

File: CatAnalyzer/src/BTagWeightEvaluator.cc (hunk 1 above)
Comment (minerva1993) [reply CatAnalyzer/src/BTagWeightEvaluator.cc]: This is done.

File: CatProducer/prod/crabConfig.py (hunk 2 above)
Comment (minerva1993) [reply CatProducer/prod/crabConfig.py]: Back to KISTI. Also I received a mail from Ryu that KISIT crab server now works okay.

File: CatProducer/python/producers/jetProducer_cfi.py (hunk 4 above)
Comment (minerva1993) [reply CatProducer/python/producers/jetProducer_cfi.py:6]: Those deep csv taggers work as CSVv2, so this works fine. Note that "pfDeepCSVJetTags:probb"+"pfDeepCSVJetTags:probbb"="pfDeepCSVDiscriminatorsJetTags:BvsAll", for Data.

File: CatAnalyzer/python/csvWeights_cfi.py (hunk 5 above)
Comment (minerva1993) [reply CatAnalyzer/python/csvWeights_cfi.py]: Made another file, deepcsvWeights_cfi.py .

**Comment by minerva1993** [comment]
Also changed broken SF csv file for CSVv2 and Deep CSV. For deepCSV, b tagging SF works fine, while CSVv2 gives some error: "ERROR in BTagCalibration: (OperatingPoint, measurementType, sysType) not available: 3, iterativefit, central13-Apr-2018" 

It's because iterative fit for CSVv2 is still under developing, "currently only available for DeepCSV, the CSVv2 payloads are being prepared". 

See: https://twiki.cern.ch/twiki/bin/viewauth/CMS/BtagRecommendation94X

**Comment by minerva1993** [comment]
We will going to make a new pull request based on v90x

**Pull request closed without merging**